Source/WebCore/html/CanvasBase.cpp

```diff
@@ -373,6 +374,203 @@ bool CanvasBase::shouldInjectNoiseBeforeReadback() const
     return scriptExecutionContext() && scriptExecutionContext()->noiseInjectionHashSalt();
 }
 
+struct RGBAColor {
```

Comment:
I'd be hesitant to add yet-another color type. Mostly this seems to be usable as uint32_t?
E.g ordering seems to be uint32_t ordering.

Alternatively you could perhaps try to use existing ColorComponents<> or something?